datastream-server-restli/src/main/java/com/linkedin/datastream/server/dms/ZookeeperBackedDatastreamStore.java

```diff
@@ -242,6 +242,18 @@ public void deleteDatastreamNumTasks(String key) {
     _zkClient.delete(path);
   }
 
+  @Override
+  public void deleteAssignmentTokens(String key) {
+    String assignmentTokensPath = KeyBuilder.datastreamAssignmentTokens(_cluster, key);
+
+    if (!_zkClient.exists(assignmentTokensPath)) {
+      LOG.info("Assignment tokens path clear for datastream {}. Nothing to delete", key);
+      return;
+    }
+
+    _zkClient.deleteRecursively(assignmentTokensPath);
```

Comment:
This method can throw an exception. Can you please check once that it will not break anything? @shrinandthakkar is evaluating the other instances of this call in the code already.

Reply:
The exception will be caught in `DatastreamResources` and propagated to the client as a rest.li service exception.